### PR TITLE
Refactoring Array#reduce into Array#findIndex

### PR DESCRIPTION
**Pull request by timdp**
The logic that finds the index of the item seemed like a good candidate for ES6's `Array.prototype.findIndex`, so I looked for a lightweight ponyfill to make the code more readable. The closest existing one I could find was `array-findindex`.

### CODE REVIEW

**Comment by jamestalmage** [comment]
I think we want to keep `loud-rejection` side-effect free. A `findIndex` equivalent that did not mutate `Array.prototype` would be better.

**Comment by jamestalmage** [comment]
After a second look it appears this doesn't do that.

LGTM.

File: api.js [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 // WARNING: This undocumented API is subject to change.
 
+// TODO Move ponyfill to separate module.
+var findIndex;
+if (Array.prototype.findIndex) {
+	findIndex = function (arr, callback, thisArg) {
+		return arr.findIndex(callback, thisArg);
+	};
```
Comment (TrySound) [inline api.js]: Isn't it equivalent of `Array.prototype.findIndex.call`?

Comment (timdp) [inline api.js]: Yeah. I was going for readability but I'd be happy to change it.

**Comment by sindresorhus** [comment]
Thank you :)

**Comment by timdp** [comment]
:cake:

**Pull request closed without merging**